optionlab/engine.py

```diff
@@ -91,30 +92,26 @@ def __init__(self, inputs: Inputs):
             else:
                 n_discarded_days = 0
 
-            self.__days2target = (
+            self.days_to_target = (
                 inputs.target_date - inputs.start_date
             ).days - n_discarded_days
         else:
-            self.__days2target = inputs.days_to_target_date
+            self.days_to_target = inputs.days_to_target_date
 
         for i, strategy in enumerate(inputs.strategy):
-            self.__type.append(strategy.type)
+            self.type.append(strategy.type)
 
             if isinstance(strategy, OptionStrategy):
-                self.__strike.append(strategy.strike)
-                self.__premium.append(strategy.premium)
-                self.__n.append(strategy.n)
-                self.__action.append(strategy.action)
-                self.__prevpos.append(strategy.prev_pos or 0.0)
+                self.strike.append(strategy.strike)
+                self._premium.append(strategy.premium)
```

Comment:
If this entire class is replaced by a function, we won't need self._premium, self._n, and so on. We can use strategy.premium, strategy.n, and so on directly. This will be more logical, certainly improving readability and reducing redundancy.